include/psen_scan_v2/monitoring_frame_deserialization.h

```diff
@@ -44,18 +50,59 @@ class FieldHeader
   Length length_;
 };
 
-using FieldId = FieldHeader::Id;
-using FieldLength = FieldHeader::Length;
-struct AdditionalFieldIds
+class MonitoringFrameFixedFields
 {
-  static constexpr FieldHeader::Id SCAN_COUNTER{ 0x02 };
-  static constexpr FieldHeader::Id MEASURES{ 0x05 };
-  static constexpr FieldHeader::Id END_OF_FRAME{ 0x09 };
+public:
+  using DeviceStatus = uint32_t;
+  using OpCode = uint32_t;
+  using WorkingMode = uint32_t;
+  using TransactionType = uint32_t;
+  using FromTheta = TenthOfDegree;
+  using Resolution = TenthOfDegree;
+
+public:
+  MonitoringFrameFixedFields(DeviceStatus device_status,
+                             OpCode op_code,
+                             WorkingMode working_mode,
+                             TransactionType transaction_type,
+                             ScannerId scanner_id,
+                             FromTheta from_theta,
+                             Resolution resolution);
+
+public:
+  DeviceStatus device_status() const;
+  OpCode op_code() const;
+  WorkingMode working_mode() const;
+  TransactionType transaction_type() const;
+  ScannerId scanner_id() const;
+  FromTheta from_theta() const;
+  Resolution resolution() const;
+
+private:
+  DeviceStatus device_status_;
+  OpCode op_code_;
+  WorkingMode working_mode_;
+  TransactionType transaction_type_;
+  ScannerId scanner_id_;
+  FromTheta from_theta_;
+  Resolution resolution_;
 };
 
+using MonitoringFrameAdditionalFieldHeaderId = MonitoringFrameAdditionalFieldHeader::Id;
+using MonitoringFrameAdditionalFieldHeaderLength = MonitoringFrameAdditionalFieldHeader::Length;
+
+namespace monitoring_frame_additional_field_header_ids
+{
+static constexpr MonitoringFrameAdditionalFieldHeaderId SCAN_COUNTER{ 0x02 };
+static constexpr MonitoringFrameAdditionalFieldHeaderId MEASURES{ 0x05 };
+static constexpr MonitoringFrameAdditionalFieldHeaderId END_OF_FRAME{ 0x09 };
+static constexpr MonitoringFrameAdditionalFieldHeaderId DIAGNOSTICS{ 0x04 };
+};  // namespace monitoring_frame_additional_field_header_ids
+
 MonitoringFrameMsg deserialize_monitoring_frame(const MaxSizeRawData& data, const std::size_t& num_bytes);
-FieldHeader readFieldHeader(std::istringstream& is, const std::size_t& max_num_bytes);
-void checkFixedFields(MonitoringFrameMsg& msg);
+MonitoringFrameFixedFields readHeader(std::istringstream& is);
+MonitoringFrameAdditionalFieldHeader readFieldHeader(std::istringstream& is, const std::size_t& max_num_bytes);
+std::vector<MonitoringFrameDiagnosticMessage> deserializeDiagnosticMessages(std::istringstream& is);
 
 class MonitoringFrameFormatError : public std::runtime_error
```

Comment:
Could be part of namespace like `psen_scan_v2::mointoring_frame`. With this namespace the name could be reduced to `FormatError`.

Reply:
We do it in another pull request